src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/SourceGenJsonTypeInfoOfT.cs

```diff
@@ -139,13 +144,12 @@ internal override void LateAddProperties()
                     return;
                 }
 
-                if (SerializeHandler != null && context?.CanUseSerializationLogic == true)
+                if (SerializeHandler == null)
```

Comment:
Can you elaborate on resolving this?

Reply:
See https://github.com/dotnet/runtime/pull/72630#discussion_r927498450

Reply:
Regarding

> I think this check was correct before and I think we should do that in other places as well.

Nothing really changes, since the `context?.CanUseSerializationLogic == true` check is also done before the call to https://github.com/dotnet/runtime/pull/72630/files#diff-43fe82ce53bba0eb09ea1dfdcd1f8774473c177831110a2cfc2a83979e9b7e11R83 on serialization.